### PR TITLE
Task2.2 string consistent anycharacters

### PR DESCRIPTION
**Pull request by YanaBocharova**
2) You are given a string allowed consisting of distinct characters and an array of strings words. A string is consistent if all characters in the string appear in the string allowed.

Return the number of consistent strings in the array words.

 

Example 1:

Input: allowed = "ab", words = ["ad","bd","aaab","baa","badab"]
Output: 2
Explanation: Strings "aaab" and "baa" are consistent since they only contain characters 'a' and 'b'.

### CODE REVIEW

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringConsistentAnycharacters.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace AlgoritmicalTasksAndUT
+{
+    public class StringConsistentAnycharacters
+    {
+        public int CountApearValues(string allowedSymb, List<string> wordsList)
+        {
+            string allowed = allowedSymb;
+
+            List<string> words = wordsList;
```
Comment (chocolatapie) [inline AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringConsistentAnycharacters.cs:14]: no need to do this, just use `wordsList`

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringContainsOnlyAllowedSymbols.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace AlgoritmicalTasksAndUT
+{
+    public class StringContainsOnlyAllowedSymbols
+    {
+        private string pattern = string.Empty;
```
Comment (chocolatapie) [inline AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringContainsOnlyAllowedSymbols.cs:11]: initialize pattern here

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringContainsOnlyAllowedSymbols.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        List<string> words = new List<string>();
```
Comment (chocolatapie) [inline AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringContainsOnlyAllowedSymbols.cs:13]: do not use this redundant `list`

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringContainsOnlyAllowedSymbols.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        List<string> listAnswrers = new List<string>();
+        public int CountWordContainsAlloedValues(string allowedSymb, List<string> wordsList)
+        {
+            pattern = $"^[{allowedSymb}]+$";
+
+            words = wordsList;
+
+            for (int i = 0; i < words.Count; i++)
+            {
+               if ( Regex.IsMatch(words[i], pattern))
```
Comment (chocolatapie) [inline AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringContainsOnlyAllowedSymbols.cs:24]: how does this regex work?

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringConsistentAnycharacters.cs (hunk 1 above)
Comment (YanaBocharova) [reply AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringConsistentAnycharacters.cs:14]: fixed

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringContainsOnlyAllowedSymbols.cs (hunk 2 above)
Comment (YanaBocharova) [reply AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringContainsOnlyAllowedSymbols.cs]: fixed

File: AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringContainsOnlyAllowedSymbols.cs (hunk 3 above)
Comment (YanaBocharova) [reply AlgoritmicalTasksAndUT/AlgoritmicalTasksAndUT/StringContainsOnlyAllowedSymbols.cs]: fixed

**Review by chocolatapie — APPROVED** [review]

**Pull request merged**